File: pegasusX/apps/driver-app-android/app/src/main/java/com/pegasusx/driver/data/telemetry/MapCameraConfig.kt
package com.pegasusx.driver.data.telemetry

/**
 * Shared driver-map camera constants — keep aligned with iOS MapCameraConfig.
 */
object MapCameraConfig {
    const val IDLE_RECENTER_MS = 8_000L
    const val INTERPOLATION_FRAME_MS = 33L
    const val CAMERA_ANIMATION_MS = 500
    const val MIN_ZOOM = 14f
    const val MAX_ZOOM = 19f
    const val NAVIGATION_TILT = 60f
    const val SPEED_ZOOM_DIVISOR = 6f
    const val LOOK_AHEAD_SECONDS = 2.0
    const val MAX_LOOK_AHEAD_METERS = 200.0
}

File: pegasusX/apps/driver-app-ios/driverappios/driverappios/Utilities/MapCameraConfig.swift
//
//  MapCameraConfig.swift
//  driverappios
//

import Foundation

/// Shared driver-map camera constants — keep aligned with Android MapCameraConfig.
enum MapCameraConfig {
    static let idleRecenterMs: UInt64 = 8_000_000_000
    static let interpolationFrameNs: UInt64 = 33_000_000
    static let cameraAnimationSeconds = 0.5
    static let minZoomDistance: Double = 400
    static let maxZoomDistance: Double = 2_500
    static let navigationPitch: Double = 60
    static let speedZoomFactor = 20.0
    static let lookAheadSeconds = 2.0
    static let maxLookAheadMeters = 200.0
}
